applications/SystemIdentificationApplication/tests/test_sensor_generator_analysis.py

```diff
@@ -0,0 +1,103 @@
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.StructuralMechanicsApplication
+import KratosMultiphysics.KratosUnittest as UnitTest
+from KratosMultiphysics.SystemIdentificationApplication.sensor_generator_analysis import SensorGeneratorAnalysis
+from KratosMultiphysics.compare_two_files_check_process import CompareTwoFilesCheckProcess
+
+class TestSensorGeneratorAnalysis(UnitTest.TestCase):
+    def test_MeshBased(self):
+        parameters = Kratos.Parameters("""{
+            "model_part_name"     : "test.ElemMat4",
+            "mdpa_file_name"      : "auxiliary_files/structure",
+            "output_file_name"    : "sensor_data.json",
+            "csv_output_file_name": "sensor_data.csv",
+            "sensor_groups"   : [
+                {
+                    "generation_method": "mesh_based",
+                    "container_type"   : "elements",
+                    "location"         : "center",
+                    "sensor_settings"  : {
+                        "name"    : "<ENTITY_ID>_dummy",
+                        "PRESSURE": 1.0,
+                        "STEP"    : 2,
+                        "VELOCITY": [1, 1, 1]
+                    }
+                }
+            ]
+        }""")
+
+        model = Kratos.Model()
+        SensorGeneratorAnalysis(model, parameters).Run()
+
+        check_params = Kratos.Parameters("""{
+            "reference_file_name"   : "auxiliary_files/sensor_data_mesh_based_ref.csv",
+            "output_file_name"      : "sensor_data.csv",
+            "remove_output_file"    : true,
+            "comparison_type"       : "deterministic",
+            "tolerance"             : 1e-6,
+            "relative_tolerance"    : 1e-9,
+            "dimension"             : 3
+        }""")
+        CompareTwoFilesCheckProcess(check_params).Execute()
+
+        check_params = Kratos.Parameters("""{
+            "reference_file_name"   : "auxiliary_files/sensor_data_mesh_based_ref.json",
+            "output_file_name"      : "sensor_data.json",
+            "remove_output_file"    : true,
+            "comparison_type"       : "deterministic",
+            "tolerance"             : 1e-6,
+            "relative_tolerance"    : 1e-9,
+            "dimension"             : 3
+        }""")
+        CompareTwoFilesCheckProcess(check_params).Execute()
+
+    def test_BoundingSurfaceBased(self):
+        parameters = Kratos.Parameters("""{
+            "model_part_name"     : "test.ElemMat2",
```

Comment:
Has it got a hole in the model part to check that it will not generate sensor in that place?

Reply:
It does not have the hole, but I changed the BB corners such that it includes an area which is not part of the model part.